service-workers/service-worker/fetch-event-async-respond-with.https.html

```diff
@@ -3,31 +3,33 @@
 <script src="resources/testharness-helpers.js"></script>
 <script src="/resources/testharnessreport.js"></script>
 <script src="resources/test-helpers.sub.js"></script>
+<script src="resources/service-worker-recorder.js"></script>
 <script>
 promise_test(function(t) {
     var script = 'resources/fetch-event-async-respond-with-worker.js';
     var scope = 'resources/simple.html';
 
     return service_worker_unregister_and_register(t, script, scope)
       .then(function(registration) {
-          return wait_for_state(t, registration.installing, 'activated');
+          return wait_for_state(t, registration.installing, 'activated')
+            .then(function() {
```

Comment:
I've structured the chains in this way whenever one callback requires a value
that is only available in the inner function's scope. I could leak that value
to the outer scope in order to keep the chain flat, but for cases like the
`registration` object, that introduces a reference that may be invalidated by
a future unregister/register cycle. So although it may be a little harder to
read, I prefer nesting closures here for safety's sake.

But you're the maintainer here, so I'm happy to defer to your preference if
that isn't convincing to you :)